core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -5117,6 +5123,47 @@ private static void computeVersion(ServletContext context) {
         }
     }
 
+    /**
+     * The default set of permissions for a new file on the platform (computed once at Jenkins startup).
+     * (provides for instance an indirect access to <code>umask</code> for Unices).
+     */
+    @Nullable
+    public static final Set<PosixFilePermission> DEFAULT_FILE_PERMISSIONS;
+
+    static {
+        LOGGER.log(Level.INFO, "Computing default file permissions");
+        boolean ok = false;
+        int maxAttempts = 10;
+        Set<PosixFilePermission> posixFilePermissions = null;
+        while (maxAttempts > 0) {
+            maxAttempts--;
+            try {
+                Path newFile = null;
+                try {
+                    newFile = Files.createFile(
+                            Paths.get("permission-getter-random-file" + RandomUtils.nextLong()));
+                    posixFilePermissions = Files.getPosixFilePermissions(newFile);
```

Comment:
throws `UnsupportedOperationException` if the associated file system does not support the PosixFileAttributeView

This needs to be caught.

Also if I am not mistaken this creates the file in the CWD of the process (which may not be writable - esecially for a service) and not in JENKINS_HOME (which may even be a different filesystem with a different permission system entirly).

Reply:
```
[ERROR] validateAntFileMaskBounded(hudson.FilePathTest)  Time elapsed: 1.431 s  <<< ERROR!
java.lang.ExceptionInInitializerError
	at hudson.ExtensionList.lookup(ExtensionList.java:418)
	at hudson.FilePath.act(FilePath.java:984)
	at hudson.FilePath.act(FilePath.java:976)
	at hudson.FilePath.mkdirs(FilePath.java:1159)
	at hudson.FilePathTest.validateAntFileMaskBounded(FilePathTest.java:553)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: java.lang.UnsupportedOperationException
	at sun.nio.fs.WindowsFileSystemProvider.readAttributes(WindowsFileSystemProvider.java:192)
	at java.nio.file.Files.readAttributes(Files.java:1737)
	at java.nio.file.Files.getPosixFilePermissions(Files.java:2004)
	at jenkins.model.Jenkins.<clinit>(Jenkins.java:5145)
	... 32 more

```